tests/searchcommands/chunked_data_stream.py

```diff
@@ -0,0 +1,103 @@
+import collections
+import csv
+import io
+import json
+
+import splunklib.searchcommands.internals
+from splunklib import six
+
+
+class Chunk(object):
+    def __init__(self, version, meta, data):
+        self.version = six.ensure_str(version)
+        self.meta = json.loads(meta)
+        self.data = csv.DictReader(io.StringIO(data.decode("utf-8")), dialect=splunklib.searchcommands.internals.CsvDialect)
+
+
+class ChunkedDataStreamIter(collections.Iterator):
+    def __init__(self, chunk_stream):
+        self.chunk_stream = chunk_stream
+
+    def __next__(self):
+        return self.next()
+
+    def next(self):
+        try:
+            return self.chunk_stream.read_chunk()
+        except EOFError:
+            raise StopIteration
+
+
+class ChunkedDataStream(collections.Iterable):
+    def __iter__(self):
+        return ChunkedDataStreamIter(self)
+
+    def __init__(self, stream):
+        empty = stream.read(0)
+        assert isinstance(empty, bytes)
+        self.stream = stream
+
+    def read_chunk(self):
+        header = self.stream.readline()
+        if len(header) == 0:
+            raise EOFError
+
+        while len(header) > 0 and header.strip() == b'':
+            header = self.stream.readline()  # Skip empty lines
+        version, meta, data = header.rstrip().split(b',')
+        metabytes = self.stream.read(int(meta))
+        databytes = self.stream.read(int(data))
+        print("sent")
+        return Chunk(version, metabytes, databytes)
+
+
+def build_chunk(kv, data=None):
+    metadata = six.ensure_binary(json.dumps(kv), 'utf-8')
+    data_output = _build_data_csv(data)
```

Comment:
IMO we should also have `six.ensure_binary` here as well.

Reply:
In this case _build_data_csv explicitly returns a binary- in fact the last line of that function is `return six.ensure_binary(csvout.getvalue())`